stratum/hal/bin/barefoot/main.cc

```diff
@@ -86,37 +80,13 @@ ::util::Status Main(int argc, char* argv[]) {
   InitGoogle(argv[0], &argc, &argv, true);
   InitStratumLogging();
 
-  char bf_sysfs_fname[128];
-  FILE* fd;
-
-  auto switchd_main_ctx = absl::make_unique<bf_switchd_context_t>();
-
-  /* Parse bf_switchd arguments */
-  CHECK_RETURN_IF_FALSE(FLAGS_bf_sde_install != "")
-      << "Flag --bf_sde_install is required";
-  switchd_main_ctx->install_dir = strdup(FLAGS_bf_sde_install.c_str());
-  switchd_main_ctx->conf_file = strdup(FLAGS_bf_switchd_cfg.c_str());
-  switchd_main_ctx->skip_p4 = true;
-  if (FLAGS_bf_switchd_background)
-    switchd_main_ctx->running_in_background = true;
-  else
-    switchd_main_ctx->shell_set_ucli = true;
-
-  /* determine if kernel mode packet driver is loaded */
-  switch_pci_sysfs_str_get(bf_sysfs_fname,
-                           sizeof(bf_sysfs_fname) - sizeof("/dev_add"));
-  strncat(bf_sysfs_fname, "/dev_add", sizeof("/dev_add"));
-  printf("bf_sysfs_fname %s\n", bf_sysfs_fname);
```

Comment:
This would also be a great opportunity to use the correct `LOG` framework.

Reply:
And replace [unsafe C string](https://stackoverflow.com/a/6904044) functions with `absl::StrFormat`.